### PR TITLE
pam_access: clean up the remote host matching code

### PR DESCRIPTION
**Pull request by t8m**
* modules/pam_access/pam_access.c (from_match): Split out remote_match()
  function and avoid calling it when matching against LOCAL keyword.

### CODE REVIEW

File: modules/pam_access/pam_access.c [hunk 1]
```diff
@@ -620,10 +621,24 @@ from_match (pam_handle_t *pamh UNUSED, char *tok, struct login_info *item)
...
 	    return (YES);
-    } else if (item->from_remote_host == 0) {	/* local: no PAM_RHOSTS */
-	if (strcasecmp(tok, "LOCAL") == 0)
-	    return (YES);
-    } else if (tok[(tok_len = strlen(tok)) - 1] == '.') {
+    } else if (strcasecmp(tok, "LOCAL") == 0) {
+	    /* LOCAL matches only local accesses */
+	    if (item->from_remote_host == 0)
+	        return YES;
+	    return NO;
```
Comment (jo-he) [inline modules/pam_access/pam_access.c:628]: This `return` is duplicate to the default below.

File: modules/pam_access/pam_access.c [hunk 2]
```diff
@@ -620,10 +621,24 @@ from_match (pam_handle_t *pamh UNUSED, char *tok, struct login_info *item)
 	if ((str_len = strlen(string)) > (tok_len = strlen(tok))
 	    && strcasecmp(tok, string + str_len - tok_len) == 0)
 	    return (YES);
-    } else if (item->from_remote_host == 0) {	/* local: no PAM_RHOSTS */
-	if (strcasecmp(tok, "LOCAL") == 0)
-	    return (YES);
-    } else if (tok[(tok_len = strlen(tok)) - 1] == '.') {
+    } else if (strcasecmp(tok, "LOCAL") == 0) {
+	    /* LOCAL matches only local accesses */
+	    if (item->from_remote_host == 0)
```
Comment (jo-he) [inline modules/pam_access/pam_access.c:626]: Maybe use `if (!item->from_remote_host)` to match the boolean nature of the next else branch.

File: modules/pam_access/pam_access.c (hunk 1 above)
Comment (t8m) [reply modules/pam_access/pam_access.c:622]: I prefer to keep it.

**Comment by t8m** [comment]
I did one more change - moved the domain match to remote match as it makes no sense to do it against TTY or SERVICE.

**Comment by t8m** [comment]
@ldv-alt Please review

File: modules/pam_access/pam_access.c (hunk 1 above)
Comment (jo-he) [reply modules/pam_access/pam_access.c:622]: Okay, but in that case
```
return item->from_remote_host ? NO : YES;
```
or at least
```
if (item->from_remote_host)
    return NO;
return YES;
```
would be a bit more readable.  ;-)

**Comment by jo-he** [comment]
closes #343

File: modules/pam_access/pam_access.c [hunk 3]
```diff
@@ -616,13 +615,28 @@ from_match (pam_handle_t *pamh UNUSED, char *tok, struct login_info *item)
...
+{
+    const char *string = item->from;
+    size_t tok_len = strlen(tok);
+    size_t str_len;
+
+    if (tok[0] == '.') {			/* domain: match last fields */
+      if ((str_len = strlen(string)) > tok_len
+	  && strcasecmp(tok, string + str_len - tok_len) == 0)
+	return YES;
     } else if (tok[(tok_len = strlen(tok)) - 1] == '.') {
```
Comment (ldv-alt) [inline modules/pam_access/pam_access.c:640]: `tok_len` is being assigned twice in this function.

**Review by ldv-alt — CHANGES_REQUESTED** [review]

File: modules/pam_access/pam_access.c (hunk 3 above)
Comment (t8m) [reply modules/pam_access/pam_access.c]: Fixed

**Review by ldv-alt — APPROVED** [review]

**Pull request merged**